nats.go

```diff
@@ -1203,18 +1220,25 @@ func (nc *Conn) connectProto() (string, error) {
 			pass, _ = u.Password()
 		}
 	} else {
-		// Take from options (pssibly all empty strings)
+		// Take from options (possibly all empty strings)
 		user = nc.Opts.User
 		pass = nc.Opts.Password
 		token = nc.Opts.Token
 	}
-	cinfo := connectInfo{o.Verbose, o.Pedantic,
-		user, pass, token,
-		o.Secure, o.Name, LangString, Version, clientProtoInfo}
+
+	cinfo := connectInfo{o.Verbose, o.Pedantic, user, pass, token,
+		o.Secure, o.Name, LangString, Version, clientProtoInfo, !o.NoEcho}
+
 	b, err := json.Marshal(cinfo)
 	if err != nil {
 		return _EMPTY_, ErrJsonParse
 	}
+
+	// Check if NoEcho is set and we have a server that supports it.
```

Comment:
Why not do this test in `processExpectedInfo()` instead?

Reply:
I could if you think that is a better place. When forming the connectProto we know we have received the server info and are processing the options to some degree, hence why I stuck it there.

Reply:
That's fine I guess, although you could do this check at top of function, since there is no point in marshalling etc.. if we are going to return an error.

Reply:
I wanted a marshal error to take precedence.

Reply:
I see.